tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,307 @@
+	require.NoError(crtb.T(), err)
+
+	crtb.client = client
+	crtb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(crtb.T(), crtb.clusterName, "Cluster name to install should be set")
+
+	crtb.clusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.clusterName)
+	require.NoError(crtb.T(), err, "Error getting cluster ID")
+	crtb.cluster, err = crtb.client.Management.Cluster.ByID(crtb.clusterID)
+	require.NoError(crtb.T(), err)
+
+	crtb.remoteClusterName = getNonLocalCluster(crtb.T(), client)
+	crtb.remoteClusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.remoteClusterName)
+	require.NoError(crtb.T(), err)
+
+	crtb.jsonData = createJsonTemplateForClusterRoleTemplateBinding().jsonData
+	require.NoError(crtb.T(), err)
+}
+
+func getNonLocalCluster(t *testing.T, client *rancher.Client) string {
+	clusterList, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(t, err)
+	var clusterName string
+	for _, clusterData := range clusterList.Data {
+		if clusterData.JSONResp["metadata"] != nil {
+			metadata, ok := clusterData.JSONResp["metadata"].(map[string]interface{})
+			if ok {
+				if nameValue, exists := metadata["name"]; exists {
+					if nameStr, isString := nameValue.(string); isString && nameStr != "local" {
+						clusterName = nameStr
+						break
+					}
+				}
+			}
+		}
+	}
+	return clusterName
+}
+
+func createJsonTemplateForClusterRoleTemplateBinding() CRTBGenTestSuite {
+	// Create a JSON object for ClusterRoleTemplateBinding template
+	crtbTemplate := map[string]interface{}{
+		"type":       "clusterRoleTemplateBinding",
+		"apiVersion": "management.cattle.io/v3",
+		"kind":       "ClusterRoleTemplateBinding",
+		"metadata": map[string]interface{}{
+			"namespace": "",
+			"name":      "",
+			"labels": map[string]interface{}{
+				"hello": "World",
+			},
+		},
+		"clusterName":       "",
+		"roleTemplateName":  "",
+		"userName":          "",
+		"userPrincipalName": "",
+	}
+
+	jsonTemplateData, _ := json.Marshal(crtbTemplate)
+
+	return CRTBGenTestSuite{jsonData: jsonTemplateData}
+}
+
+func unmarshalReadJson(t *testing.T, readJson []byte, remoteClusterID string, crtbName string, newUser string, roleTemplateName string) map[string]interface{} {
+	// Unmarshal JSON object for ClusterRoleTemplateBinding template
+	var inputData map[string]interface{}
+	err := json.Unmarshal(readJson, &inputData)
+	require.NoError(t, err)
+	inputData["metadata"].(map[string]interface{})["namespace"] = remoteClusterID
+	inputData["clusterName"] = remoteClusterID
+	inputData["metadata"].(map[string]interface{})["name"] = crtbName
+	inputData["userName"] = newUser
+	inputData["roleTemplateName"] = roleTemplateName
+	return inputData
+}
+
+func validateCrtbCreationBySteveWithAllRequiredInputValues(t *testing.T, client *rancher.Client, remoteClusterID string, roleTemplateName string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
```

Comment:
does this mean, until cleanup, we will have `n` number of users created where n=number of validate functions in this test? Is that expected?

Reply:
I moved it to setup suite.